src/Confluent.Kafka/Consumer.cs

```diff
@@ -430,13 +430,15 @@ public Consumer(IEnumerable<KeyValuePair<string, object>> config)
                     defaultTopicConfig.ToList().ForEach((kvp) => { topicConfigHandle.Set(kvp.Key, kvp.Value.ToString()); });
                 }
                 LibRdKafka.conf_set_default_topic_conf(configPtr, topicConfigHandle.DangerousGetHandle());
+                topicConfigHandle.SetHandleAsInvalid(); // topic config object is no longer useable.
             }
 
             LibRdKafka.conf_set_error_cb(configPtr, ErrorCallback);
             LibRdKafka.conf_set_log_cb(configPtr, LogCallback);
             LibRdKafka.conf_set_stats_cb(configPtr, StatsCallback);
 
             this.kafkaHandle = SafeKafkaHandle.Create(RdKafkaType.Consumer, configPtr);
+            configHandle.SetHandleAsInvalid(); // config object is no longer useable.
```

Comment:
The passed conf_t is only invalidated if rd_kafka_new() succeeded, if it returns error the caller will need to destroy it.